dao/src/main/java/org/thingsboard/server/dao/sql/query/EntityDataAdapter.java

```diff
@@ -92,8 +92,9 @@ static String convertValue(Object value) {
                 }
                 try {
                     double dblVal = Double.parseDouble(strVal);
-                    if (!Double.isInfinite(dblVal)) {
-                        return Double.toString(dblVal);
+                    String doubleAsString = Double.toString(dblVal);
```

Comment:
Hey! @ShvaykaD 

Look at a more optimal way to identify a double type (have a `.` char).
Another improvement is to verify the string length to asses for long or double string content (let's say 64 chars)
I hope you can add more use cases for conversion as well.
It is essential code that affects data representation.
![image](https://user-images.githubusercontent.com/79898499/215509070-21236aed-25b1-4978-b04b-96bc7377ca08.png)